cmd/object-api-utils.go

```diff
@@ -533,14 +532,21 @@ func NewGetObjectReader(rs *HTTPRangeSpec, oi ObjectInfo, pcfn CheckCopyPrecondi
 		}
 	}()
 
-	isEncrypted := crypto.IsEncrypted(oi.UserDefined)
-	isCompressed, err := oi.IsCompressedOK()
-	if err != nil {
-		return nil, 0, 0, err
+	var (
+		isEncrypted  = crypto.IsEncrypted(oi.UserDefined)
+		isCompressed = oi.IsCompressed()
+	)
+	if isEncrypted && isCompressed {
```

Comment:
This is never possible @aead - why do we need this? shouldn't we fix the cause than this defensive code?

Reply:
This is to keep the old behavior - to some extend. Before, we used `oi.IsCompressedOK` which did this check implicitly. I made it explicit for the exactly this argument.

However, I'm also in favor of removing it. If at all, such checks should happen at the top (as high up the stack as possible) once and not done over and over again in (more) low-level calls.

My only concern is that I want change not too much behavior to keep PRs reviewable. However, if reviewers are okay with it I can also remove the check.